### ISSUE TITLE
Scribe sends integers as strings in JSON?

### ISSUE THREAD

**Issue by alexweb** [knuckleswtf/scribe#640]
Hi everyone.
Looks like scribe doesn't work properly with integer type when sending JSON? 
My annotations look like this (see `id` type): 
```
    /**
     * @group Coupon
     *
     * Update Coupon
     * @authenticated
     * @bodyParam data object required
     * @bodyParam data.id int required Example: 1
     * @bodyParam data.name string required Example: COUPONNAME
     * @bodyParam data.percentage int required Example: 30
     * @bodyParam data.permanent bool required Example: true
     * @bodyParam data.startDate string optional Example: 2023-01-05T14:48:00.000Z
     * @bodyParam data.endDate string optional Example: 2023-10-05T14:48:00.000Z
     * @throws Error422Exception|Throwable
     */
```

Resulting request: 
```
{
  "id": "1",
  "name": "COUPONNAME",
  "percentage": "30",
  "permanent": true,
  "startDate": "2023-01-05T14:48:00.000Z",
  "endDate": "2023-10-05T14:48:00.000Z"
}
```

Looks like Scribe still wraps integers into quotes? 

Or is there any option to change this behaviour? I would like it to send integer as integer instead (see `id` and `percentage`): 

```
{
  "id": 1,
  "name": "COUPONNAME",
  "percentage": 30,
  "permanent": true,
  "startDate": "2023-01-05T14:48:00.000Z",
  "endDate": "2023-10-05T14:48:00.000Z"
}
```

<img width="1222" alt="Screenshot 2023-03-28 at 12 48 03" src="https://user-images.githubusercontent.com/106749173/228199654-755d11ac-a260-4a03-8339-90a974989061.png">

**Comment by shalvah** [comment]
It's probably because when you're reading a text input in HTML, every value is a string (and Scribe doesn't do any conversion). The fix would probably be to render a `number` input for number types (rather than a `text` input). See https://github.com/knuckleswtf/scribe/blob/b0d2bbe20e641e7a6d9649481e454b1dc166b584/resources/views/components/field-details.blade.php#L25 — we currently render booleans specially, but not numbers. PR welcome.

**Comment by alexweb** [comment]
Thanks for the reply. Will need to fix this

**Comment by shalvah** [comment]
Okay, I'm wrong. We do set it.

https://github.com/knuckleswtf/scribe/blob/195c89786d90f1423739f533ba6bfa33e20a159b/resources/views/components/field-details.blade.php#L19-L23

I think it's the HTML/JS. IIRC, reading an input value in JS always gives you a string, even if the input type is number.

**Comment by mselmanyildirim** [comment]
Do we have a workaround to send booleans from Scribe UI?   
Similar to reported above, when I generate the document via Laravel's validation as follows:

![image](https://github.com/knuckleswtf/scribe/assets/61732943/8250c8c2-ddac-4f12-bbef-95d21ae04a64)

and select the parameter via Scribe API UI:

![image](https://github.com/knuckleswtf/scribe/assets/61732943/1dba948a-0941-4d22-b957-17b8cea6bd81)

I get the following error:

![image](https://github.com/knuckleswtf/scribe/assets/61732943/2178c156-f3d9-4fb0-bd71-89676c6caf79)

When I try to debug with the following:

![image](https://github.com/knuckleswtf/scribe/assets/61732943/84b98839-f281-4ed0-b995-96f79c2f59b4)

I get this result, saying it is not boolean, but rather string:

![image](https://github.com/knuckleswtf/scribe/assets/61732943/cbe1316e-1844-4d1c-baab-0e470c77a68d)
 
What I finally did as a workaround is adding the type as Enum in the codeblock as the following:

![image](https://github.com/knuckleswtf/scribe/assets/61732943/1c79892a-5b6d-468e-8ba1-657e2df2ccc1)

And Scribe UI becomes like:

![image](https://github.com/knuckleswtf/scribe/assets/61732943/3beef312-db3a-489d-b884-1b727775151b)

It still gets sent as string, but at least Laravel validates it to be a boolean. It is not very handsome comparing to radio switch though 😉

**Comment by shalvah** [comment]
> IIRC, reading an input value in JS always gives you a string

**Comment by shalvah** [comment]
PRs appreciated

**Comment by ptelevich** [comment]
@shalvah got it, https://github.com/knuckleswtf/scribe/pull/830

**Comment by shalvah** [comment]
fixed in 4.35.0
